classes/Builder/Payload/OrderPayloadBuilder.php

```diff
@@ -0,0 +1,299 @@
+            'invoice_id' => '',
+            'description' => $this->truncate('Checking out with your cart from ' . \Configuration::get('PS_SHOP_NAME'), 127),
+            'soft_descriptor' => $this->truncate(\Configuration::get('PS_SHOP_NAME'), 22),
+            'amount' => [
+                'currency_code' => $this->cart['currency']['iso_code'],
+                'value' => $this->cart['cart']['totals']['total_including_tax']['amount'],
+            ],
+            'shipping' => [
+                'name' => [
+                    'full_name' => 'Mr ' . $this->cart['addresses']['shipping']->lastname . ' ' . $this->cart['addresses']['shipping']->firstname,
+                ],
+                'address' => [
+                    'address_line_1' => $this->cart['addresses']['shipping']->address1,
+                    'address_line_2' => $this->cart['addresses']['shipping']->address2,
+                    'admin_area_1' => (string) $this->getStateNameById($this->cart['addresses']['shipping']->id_state),
+                    'admin_area_2' => $this->cart['addresses']['shipping']->city,
+                    'country_code' => $countryCodeMatrice->getPaypalIsoCode($shippingCountryIsoCode),
+                    'postal_code' => $this->cart['addresses']['shipping']->postcode,
+                ],
+            ],
+            'payer' => [
+                'name' => [
+                    'given_name' => $this->cart['customer']->lastname,
+                    'surname' => $this->cart['customer']->firstname,
+                ],
+                'email_address' => $this->cart['customer']->email,
+                'address' => [
+                    'address_line_1' => $this->cart['addresses']['invoice']->address1,
+                    'address_line_2' => $this->cart['addresses']['invoice']->address2,
+                    'admin_area_1' => (string) $this->getStateNameById($this->cart['addresses']['invoice']->id_state), //The highest level sub-division in a country, which is usually a province, state, or ISO-3166-2 subdivision.
+                    'admin_area_2' => $this->cart['addresses']['invoice']->city, // A city, town, or village. Smaller than admin_area_level_1
+                    'country_code' => $countryCodeMatrice->getPaypalIsoCode($payerCountryIsoCode),
+                    'postal_code' => $this->cart['addresses']['invoice']->postcode,
+                ],
+            ],
+            'payee' => [
+                'merchant_id' => (new PaypalAccountRepository())->getMerchantId(),
+            ],
+            'application_context' => [
+                'brand_name' => \Configuration::get('PS_SHOP_NAME'),
+                'shipping_preference' => 'SET_PROVIDED_ADDRESS',
+            ],
+            'roundingConfig' => (string) \Configuration::get('PS_ROUND_TYPE') . '-' . (string) \Configuration::get('PS_PRICE_ROUND_MODE'),
+        ];
+
+        $this->getPayload()->setItems($node);
```

Comment:
You may move this call at the end of the method, and avoid duplicate it in each `if`.

Reply:
Removed :)